va/caa.go

```diff
@@ -64,8 +64,8 @@ func (va *ValidationAuthorityImpl) checkCAA(
 		validationMethod = params.validationMethod
 	}
 
-	va.log.AuditInfof("Checked CAA records for %s, [Present: %t, Account ID: %s, Challenge: %s, Valid for issuance: %t] Records=%s",
-		identifier.Value, present, accountID, validationMethod, valid, recordsStr)
+	va.log.AuditInfof("Checked CAA records for %s, [Present: %t, Account ID: %s, Challenge: %s, Valid for issuance: %t] Records=%s Response=%q",
```

Comment:
The new `response` field should contain all the same information as `recordsStr`, plus some more, so it can replace the `Records=` part of the log line.

Reply:
Alright, so there are two ways of doing this:

1. No longer return `records` from the `va.parseResults` when `va.checkCAARecords` is called. This breaks `TestParseResults`, we could do a re-write that tests the string literal of `response` for the full dig-like response
2. Assign `records` returned from `va.parseResults` to `_` in the actual code, but assign it to `records`. This preserves the existing `TestParseResults` and it's easier to test a slice of just the records we care about

Reply:
I'm a bit confused. I think I was just asking for a change in the log output. Does that impact how the tests run?

Reply:
Yeah, and I may just be wrong here too:

`recordsStr` will be removed the only place it's used:
https://github.com/letsencrypt/boulder/blob/e8ca1987cd303affda2fb1689834fb4941e50f32/va/caa.go#L67-L68

If `recordsStr` is removed, the JSON marshal is removed here:
https://github.com/letsencrypt/boulder/blob/e8ca1987cd303affda2fb1689834fb4941e50f32/va/caa.go#L54-L57

If the JSON marshal goes away, then `records`, which is assigned here, is never used:
https://github.com/letsencrypt/boulder/blob/e8ca1987cd303affda2fb1689834fb4941e50f32/va/caa.go#L49-L52

So following `records` back to `va.checkCAARecords` we see it's assigned when `va.getCAASet` is called and is never used other than when it's returned to `va.checkCAA`:
https://github.com/letsencrypt/boulder/blob/e8ca1987cd303affda2fb1689834fb4941e50f32/va/caa.go#L195-L200

Following it back to the call to getCAASet, `records` is parsed from a `caaResult` object in the  `results` slice:
https://github.com/letsencrypt/boulder/blob/e8ca1987cd303affda2fb1689834fb4941e50f32/va/caa.go#L122-L124
https://github.com/letsencrypt/boulder/blob/e8ca1987cd303affda2fb1689834fb4941e50f32/va/caa.go#L127-L135

And returned by `va.getCAASet` -> `va.parseResults`:
https://github.com/letsencrypt/boulder/blob/e8ca1987cd303affda2fb1689834fb4941e50f32/va/caa.go#L158-L172

In `caa_test.go` records is used to validate that `va.parseResults` is working properly in test `TestParseResults`:
https://github.com/letsencrypt/boulder/blob/e8ca1987cd303affda2fb1689834fb4941e50f32/va/caa_test.go#L591-L613

Reply:
Keeping this around in the return for `va.parseResults` just for testing feels wonky to me, but the alternative doesn't seem great either.

Reply:
> 2. Assign records returned from va.parseResults to _ in the actual code, but assign it to records
2. Assign records returned from `va.parseResults` to `_` in the actual code, but assign it to `records` in our Tests

Also, typo, sorry for the confusion there.

Reply:
> So following records back to va.checkCAARecords we see it's assigned when va.getCAASet is called and is never used other than when it's returned to va.checkCAA:

`records` in this call chain is doing the same thing as the new `results` return value you added: providing context to be logged. So, every place that was/is returning `records` should now be returning `results`.

I wouldn't worry about the fact that TestParseResults checks the value of that return parameter. Since you're refactoring the function to return something different, you should expect to also refactor the test to check for that different thing.

Reply:
Alright, I think we're on the same page. Thanks!